### ISSUE TITLE
Could ScenarioCollection iterate through user defined combinations?

### ISSUE THREAD

**Issue by jetuk** [pywr/pywr#385]
Currently `ScenarioCollection` iterates through the product of all scenario ensembles. It would be useful if the user could specify their own combinations to run either for debugging or just analysis on a subset. 

What will break if this changes? Presumably `len(model.scenarios.combinations)` is used a lot to check the correct size of things that vary by scenario. In this proposal there would be two lengths: the maximum possibly and the number to run. These could then be different.

Thoughts?

**Comment by snorfalorpagus** [comment]
I think this could be useful, especially if you could use slices for each scenario.

model.scenarios[0:50,(1,2)]

I don't think this should be too difficult to make transparent to most of the existing code.

**Comment by snorfalorpagus** [comment]
Turns out `model.scenario.combinations` is just a list.

https://github.com/pywr/pywr/blob/master/pywr/_core.pyx#L73-L79

It should be trivial to filter this list as it's created. Any suggestions for a syntax? I have a use case for this right now.

**Comment by jetuk** [comment]
I think there's two ways to setup,

1. With a truncated range of each scenario.
2. A predefined list of tuples to run.

JSON format for (1) could be like so. Where slice could be an integer for a single scenario or anything that could be passed to `slice`. We would have to parse a string for that I think.
```
"scenarios": [
  { "name": "A", "size": 10, "slice": ":2"},
  { "name": "B", "size": 4, "slice": "0"},
]
```
For (2) you could provide an optional list of lists. The one below would only run 4 combinations. 

```
"scenario-combinations": [
  [0, 0],
  [0, 2],
  [4, 0],
  [4, 2],
]
```
I think these could be passed to the `ScenarioCollection` and if no predefined combinations are set it iterates through per scenario defined slices if available, otherwise does everything.

**Comment by snorfalorpagus** [comment]
https://github.com/pywr/pywr/commit/2902452222fdac420cff4f4e2df110b2410dd3b8

Looks like `ScenarioCombinations` isn't used but was never removed.

**Comment by jetuk** [comment]
Closed by #387
